bundle/edu.gemini.spModel.smartgcal/src/main/java/edu/gemini/spModel/smartgcal/provider/CalibrationProviderImpl.java

```diff
@@ -33,18 +33,30 @@ public CalibrationProviderImpl(CalibrationRepository repository) {
 
     @Override
     public List<Calibration> getCalibrations(CalibrationKey key) {
-        ConfigurationKey configKey = key.getConfig();
+        final ConfigurationKey configKey = key.getConfig();
+        final List<Calibration> cals;
         if (key instanceof CalibrationKeyImpl.WithWavelength) {
-            Double wavelength = ((CalibrationKeyImpl.WithWavelength) key).getWavelength();
-            return getCalibrations(
-                    configKey.getInstrumentName(),
-                    configKey,
-                    wavelength);
+            final Double wavelength = ((CalibrationKeyImpl.WithWavelength) key).getWavelength();
```

Comment:
Not that it's really important, but I suppose there's no reason to use `Double` instead of `double` here, especially since this is unboxed in the called method?

I'm not even sure why `CalibrationKeyImpl.WithWavelength` has a `Double` instead of a `double`, since a quick glance at the code that instantiates these objects in the instrument specifically seems to make sure `null` will never be passed into the constructor for the wavelength value.